src/Avalonia.Base/Compatibility/OperatingSystem.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Avalonia.Compatibility
+{
+    internal sealed class OperatingSystemEx
+    {
+#if NET6_0_OR_GREATER
+        public static bool IsWindows() => OperatingSystem.IsWindows();
+        public static bool IsMacOS() => OperatingSystem.IsMacOS();
+        public static bool IsLinux() => OperatingSystem.IsLinux();
+        public static bool IsAndroid() => OperatingSystem.IsAndroid();
+        public static bool IsIOS() => OperatingSystem.IsIOS();
+        public static bool IsBrowser() => OperatingSystem.IsBrowser();
+        public static bool IsOSPlatform(string platform) => OperatingSystem.IsOSPlatform(platform);
+#else
+        public static bool IsWindows() => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        public static bool IsMacOS() => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        public static bool IsLinux() => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+        public static bool IsAndroid() => IsOSPlatform("ANDROID");
+        public static bool IsIOS() => IsOSPlatform("IOS");
+        public static bool IsBrowser() => IsOSPlatform("BROWSER");
```

Comment:
You can also make them eliminable by wrapping readonly static fields, for example:

```csharp
private readonly static bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
public static bool IsWindows() => isWindows;
```

This will make the `isWindows` a runtime constant even on those TFMs dont't have `OperatingSystem.*` APIs.

Reply:
I don't think it really matters, as trimming in .NET wasn't as good before .NET 5/6. I would prefer keeping old-tfms code as simple as possible.